metrics/cgroups/v1/memory.go

```diff
@@ -543,7 +543,7 @@ var memoryMetrics = []*metric{
 		unit: metrics.Total,
 		vt:   prometheus.GaugeValue,
 		getValues: func(stats *v1.Metrics) []value {
-			if stats.Memory == nil {
+			if stats.Memory == nil || stats.Memory.Usage == nil {
```

Comment:
maybe we can use `stats.GetMemory().GetUsage()`?

Reply:
This is a good idea, but the behavior changes a bit when `stats.Memory == nil || stats.Memory.Usage == nil` , and I'm not sure if that's what we expect

https://github.com/containerd/containerd/blob/878132923db35f31c8c7b324d9cd1e5f59046064/metrics/cgroups/v1/metric.go#L49-L52

Reply:
I think the condition is to check the usage is not nil so that the `return` can use the value from usage.

Reply:
Sorry for my misunderstanding, this looks great

Reply:
thx guys! updated